src/index.ts

```diff
@@ -178,17 +181,19 @@ function buildURL(url: URL): string {
 
 export class Connection {
   private config: Config
+  private fetch: Fetch
   private session: QuerySession | null
   private url: string
 
   constructor(config: Config) {
-    this.session = null
-    this.config = { ...config }
-
-    if (typeof fetch !== 'undefined') {
-      this.config.fetch ||= fetch
+    if (!config.fetch && typeof fetch === 'undefined') {
+      throw new Error('No `fetch` implementation available')
```

Comment:
To me, this is the only thing that seems like change in behavior.

So i understand, before if there was no fetch passed into the config and no global fetch available, we'd have not failed in the constructor, but failed later inside of postJSON when we tried to execute the undefined thing. Correct?

But our change here catches this in the constructor and yells before trying to use it.